src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -376,8 +375,17 @@ private static string[] ParseArgv(ProcessStartInfo psi, string alternativePath =
                     argvList.Add("openURL"); // kfmclient needs OpenURL
                 }
             }
+
             argvList.Add(argv0);
-            ParseArgumentsIntoList(psi.Arguments, argvList);
+            if (!string.IsNullOrEmpty(psi.Arguments))
+            {
+                ParseArgumentsIntoList(psi.Arguments, argvList);
+            }
+            else
+            {
+                argvList.AddRange(psi.ArgumentList);
+                psi.Arguments = PasteArguments.Paste(psi.ArgumentList, true);
```

Comment:
I don't think we want to mutate the psi.Arguments like this. a) It's not functionally necessary, but more importantly b) it'll then prevent use of this ProcessStartInfo object again, as the next time the dev tries to use it, both the Arguments and ArgumentsList properties will be set and Start will throw.

Reply:
While the original proposal called for the two properties to mirror updates to each other automatically, the thinking was that this would be done on a lazy basis so that we don't do unnecessary conversions. 

I do not recall whether we kept the mirroring idea or whether we decided to leave `ProcessStartInfo` as a "dumb bag of data" (or whether that was even discussed. Api reviews tend to focus only on shape, not semantics.)

Either way, we should make a firm decision on this before committing code.